### PR TITLE
Support/qaa 259

### PR DESCRIPTION
**Pull request by VicAlbr**
### ✅ Checklist

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

Adding delegate test

### ❓ Context

- **JIRA or GitHub link**: [QAA-259](https://ledgerhq.atlassian.net/browse/QAA-259)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[QAA-259]: https://ledgerhq.atlassian.net/browse/QAA-259?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/tests/page/modal/delegate.modal.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { step } from "tests/misc/reporters/step";
+
+export class delegateModal extends Modal {
+  private titleProvider = this.page.getByTestId("modal-provider-title");
+  private delegateContinueButton = this.page.locator("id=delegate-continue-button");
+  private rowProvider = this.page.getByTestId("modal-provider-row");
+  private stakeProviderContainer = (stakeProviderID: string) =>
+    this.page.getByTestId(`stake-provider-container-${stakeProviderID}`);
+
+  async getTitleProvider() {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-desktop/tests/page/modal/delegate.modal.ts:12]: can you please add @step descriptions for every method in this new class?

File: apps/ledger-live-desktop/tests/page/modal/delegate.modal.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  private stakeProviderContainer = (stakeProviderID: string) =>
+    this.page.getByTestId(`stake-provider-container-${stakeProviderID}`);
+
+  @step("Get title provider")
+  async getTitleProvider() {
+    await this.titleProvider.waitFor();
+    return await this.titleProvider.textContent();
+  }
+
+  @step("Verify provider")
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-desktop/tests/page/modal/delegate.modal.ts:21]: You can add provider name in the description like "Verify provider is $0"

File: apps/ledger-live-desktop/tests/page/modal/delegate.modal.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  async continueDelegate() {
+    await this.delegateContinueButton.click();
+  }
+
+  @step("Click on search provider button")
+  async openSearchProviderModal() {
+    await this.searchOpenButton.click();
+  }
+
+  @step("Input provider")
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-desktop/tests/page/modal/delegate.modal.ts:39]: same here

File: apps/ledger-live-desktop/tests/page/modal/delegate.modal.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  async inputProvider(provider: string) {
+    await this.inputSearchField.fill(provider);
+  }
+
+  @step("Select provider")
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-desktop/tests/page/modal/delegate.modal.ts:44]: here

File: apps/ledger-live-desktop/tests/page/modal/delegate.modal.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  async selectProvider(providerIndex: number) {
+    await this.rowProvider.nth(providerIndex).click();
+    await this.searchCloseButton.click();
+  }
+
+  @step("Click on chosen stake provider")
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-desktop/tests/page/modal/delegate.modal.ts:50]: and here

**Review by ypolishchuk-ledger — APPROVED** [review]

**Review by LucasWerey — APPROVED** [review]

**Pull request merged**